docs/src/man/doctests.md

```diff
@@ -105,59 +102,101 @@ line where checking should stop, i.e.
 julia> div(1, 0)
 ERROR: DivideError: integer division error
 [...]
+```
+````
+
+## Preserving definitions between blocks
 
+Every doctest block is evaluated inside its own `module`. This means that definitions
+(types, variables, functions etc.) from a block can *not* be used in the next block.
+For example:
+
+````markdown
+```jldoctest
+julia> foo = 42
+42
 ```
 ````
 
-## Skipping Doctests
+In the doctest block below The variable `foo` will not be defined in the next block:
 
-Doctesting can be disabled by setting the [`makedocs`](@ref) keyword `doctest = false`.
-This should only be done when initially laying out the structure of a package's
-documentation, after which it's encouraged to always run doctests when building docs.
+```jldoctest
+julia> println(foo)
+ERROR: UndefVarError: foo not defined
+```
+````
+
+To preserve definitions it is possible to label blocks in order to collect several blocks
+into the same module. All blocks with the same label (in the same file) will be evaluated
+in the same module, and hence share scope. This can be useful if the same definitions are
+used in more than one block, with for example text, or other doctest blocks, in between.
+Example:
+
+````markdown
+```jldoctest mylabel
+julia> foo = 42
+42
+```
+````
+
+Now, since the block below has the same label as the block above, the variable `foo` can
+be used:
+
+````markdown
+```jldoctest mylabel
+julia> println(foo)
+42
+```
+````
+
+!!! note
+
+    Labeled doctest blocks does not need to be consecutive (as in the example above) to be
+    included in the same module. They can be interspaced with unlabeled blocks or blocks
+    with another label.
 
 ## Setup Code
 
 Doctests may require some setup code that must be evaluated prior to that of the actual
-example, but that should not be displayed in the final documentation. It could also be that
-several separate doctests require the same definitions. For both of these cases a `@meta`
-block containing a `DocTestSetup = ...` value can be used as follows:
+example, but that should not be displayed in the final documentation. For this purpose a
+`@meta` block containing a `DocTestSetup = ...` value can be used as follows:
 
-    ```jldoctest
-    julia> using DataFrames
+````markdown
```

Comment:
Something going on here as well. I guess this bit can be dropped and the two paragraphs should be merged?